pytesmo/validation_framework/adapters.py

```diff
@@ -39,40 +39,93 @@
 class MaskingAdapter(object):
     """
     Transform the given class to return a boolean dataset given the operator
-    and threshold. This class calls the callse the read_ts and read methods
+    and threshold. This class calls the read_ts and read methods
     of the given instance and applies boolean masking to the returned data
     using the given operator and threshold.
 
     Parameters
     ----------
     cls: object
-        has to have read_ts method
+        has to have a read_ts or read method
     operator: string
-        one of '<', '<=', '==', '>=', '>'
-    threshold: float
+        one of '<', '<=', '==', '>=', '>', '!='
+    threshold:
         value to use as the threshold combined with the operator
-
+    column_name: string, optional
+        name of the column to cut the read masking dataset to
     """
 
-    def __init__(self, cls, op, threshold):
+    def __init__(self, cls, op, threshold, column_name = None):
         self.cls = cls
 
         self.op_lookup = {'<': operator.lt,
                           '<=': operator.le,
                           '==': operator.eq,
+                          '!=': operator.ne,
                           '>=': operator.ge,
                           '>': operator.gt}
         self.operator = self.op_lookup[op]
         self.threshold = threshold
 
+        self.column_name = column_name
+
+    def __mask(self, data):
+        if self.column_name is not None:
+            data = data.loc[:, [self.column_name]]
+        return self.operator(data, self.threshold)
+
     def read_ts(self, *args, **kwargs):
         data = self.cls.read_ts(*args, **kwargs)
-        return self.operator(data, self.threshold)
+        return self.__mask(data)
 
     def read(self, *args, **kwargs):
         data = self.cls.read(*args, **kwargs)
-        return self.operator(data, self.threshold)
+        return self.__mask(data)
+
+class SelfMaskingAdapter(object):
```

Comment:
Could you add a small paragraph to the documentation about the `SelfMaskingAdapter` and the differences to the `MaskingAdapter`. I think that would help us in making it clearer what they both do. I think we might be able to combine them into one class since they are very similar. What do you think?

Reply:
Re documentation, sure, as soon as we've sorted out what we actually want to do. Re combining: pro: avoiding code repetition, contra: both adapters do something different and they yield a quite different result. My personal taste as a dev would be to have two separate classes for two separate concerns - but I could derive them from a common base class to avoid code repetition. If I do that, should I use an https://docs.python.org/2/library/abc.html ?

Reply:
For the documentation I think we should edit the ipython notebooks and automatically include them in the sphinx documentation. For now editing https://github.com/TUW-GEO/pytesmo/blob/master/docs/validation_framework.ipynb and exporting it as rst should also work.

Reply:
I think we can keep the classes separate for now.

Reply:
OK, I've updated the docs - what do you think?

Reply:
Looks good, I'll merge this after a quick rebase if all the tests still pass.